CleanSweep_Group6/src/main/java/edu/depaul/cleanSweep/controlSystem/Cleaner.java

```diff
@@ -31,6 +32,8 @@
 	// In order to traverse through the history, start at the head, and work downward
 	private List<Pair<Integer, SurfaceType>> vacuumBag = new LinkedList<Pair<Integer, SurfaceType>>();
 
+	private ArrayList<FloorTile> cleanerHistory = new ArrayList<FloorTile>();
+
```

Comment:
I don't necessarily think its a bad idea to use an arraylist for storing the cleaner movement history since you're storing each of the floortile nodes inside, however when it comes time to re-traverse this list for like a charging station, I think that an arraylist may make things more difficult. I mean, you may have a different idea of how this might work but In my head, I'm thinking that with an arraylist youll be storing multiple of the same node since you're just adding them to the list each visit. Maybe a scalability concern? Also, genuine question, say we want to pull the last node on the list since it was the most recent and we want to get back to the nearest charging station, what algorithm could we use so that we aren't going the exact same path we went before and visiting tiles multiple times.

My vision, not that you have to do this at all, of course, was that you would be able to use my custom linked list or something similar to create a linked list that the cleaner would store and would know its own links. For instance, you could add a separate 'add' method in the customlinkedlist that just adds nodes as they appear 

Ex. I'm at (0,0) add node link from list-head south 1 space (1,0), add node link east 1 space (1,1), add node west 1 space.
(arrows indicate links, always when moving link to your previous node)
(0,0)
   ^
   |
   v
(1,0) <--> (1,1)

Now as the cleaner I would know that if I wanted to get back to 0,0 from 1,1 I can go west-north

Reply:
I see where you're coming from. But an arraylist stores the indecies. So if you want to know "what was the fifth move I did". You can jump right to that. No need to traverse counting up to 5 each time. In addition, I think this is what we want. Because say the robot moves over a tile, and then moves over the tile later in its life. That first traversal is stored at index (or "time" if you prefer) 5. When it crosses over again, another copy is stored in at time 15. This gives us an exact path, and in order to traverse backwards, you only need to move from array.length -> 0 which is equivalent to going backwards in time. I don't really understand why you want me to use a custom linked list when its already built into java. In addition, if you're really concerned and want to use it as a linked list, use javastreams to convert the arraylist to a linked list.